assets/prototype/application/services/apollo/mutations/useEntityMutation.ts

```diff
@@ -1,39 +1,55 @@
 import { useApolloClient } from '@apollo/react-hooks';
-import { ApolloError } from 'apollo-client';
+import { ApolloError, MutationUpdaterFn, MutationOptions, OperationVariables } from 'apollo-client';
+import { DataProxy } from 'apollo-cache';
+import { FetchResult } from 'apollo-link';
 import { useState } from '@wordpress/element';
 import { pathOr } from 'ramda';
 
 import { mutations } from '../../../../domain/eventEditor/data/mutations';
 import useMutators from './useMutators';
 import useIfMounted from '../../../hooks/useIfMounted';
+import {
+	EntityMutation,
+	OnMutationCompletedFn,
+	OnUpdateFn,
+	OnMutationErrorFn,
+	MutationInput,
+	MutationResult,
+	CustomMutationOptions,
+	Mutator,
+	MutationType,
+	EntityType,
+} from './types';
+
+const DEFAULT_RESULT: MutationResult = { called: false, loading: false };
 
 /**
  * @param {string} type Entity type name
  * @param id
  */
-const useEntityMutation = (type, id = '') => {
+const useEntityMutation = (type: EntityType, id: string = ''): EntityMutation => {
 	const client = useApolloClient();
-	const [result, setResult] = useState({ called: false, loading: false });
+	const [result, setResult] = useState(DEFAULT_RESULT);
 	const mutators = useMutators();
 	const ifMounted = useIfMounted();
 
 	/**
 	 * @param {string} mutationType Type of mutation - CREATE|UPDATE|DELETE
 	 */
-	const getMutation = (mutationType = 'UPDATE') => {
+	const getMutation = (mutationType: MutationType = MutationType.Update): any => {
 		// For example "CREATE_DATETIME"
-		const mutation = `${mutationType}_${type.toUpperCase()}`;
+		const mutation: string = `${mutationType}_${type.toUpperCase()}`;
```

Comment:
`: string` is kind of optional in this case since `MutationType` and `EntityType` are enums of strings.